python/graphstorm/inference/mt_infer.py

```diff
@@ -309,13 +311,22 @@ def nfrecon_gen_embs(skip_last_self_loop=False, node_embs=embs):
                                                                   inplace=True)
 
                     decoder = model.task_decoders[task_info.task_id]
-                    ranking = run_lp_mini_batch_predict(decoder, lp_test_embs, dataloader, device)
-                    pre_results[task_info.task_id] = ranking
+                    ranking, test_lengths = run_lp_mini_batch_predict(
+                        decoder, lp_test_embs, dataloader, device, return_batch_lengths=True)
+                    pre_results[task_info.task_id] = (ranking, test_lengths)
 
         if do_eval:
             test_start = time.time()
+            assert isinstance(self.evaluator, GSgnnMultiTaskEvaluator)
+
             val_score, test_score = self.evaluator.evaluate(
-                pre_results, pre_results, 0)
+                pre_results,
+                pre_results,
+                0,
+                val_candidate_sizes=test_lengths,
```

Comment:
Do we need to send `val_candidate_sizes` and `test_candidate_sizes` to `GSgnnMultiTaskEvaluator.evaluate`?
You have sent them through pre_results.

Reply:
Maybe I'm confusing the interface here, let me see if we need both.